x-pack/plugins/spaces/server/capabilities/capabilities_switcher.ts

```diff
@@ -54,41 +54,63 @@ function toggleDisabledFeatures(
 ) {
   const disabledFeatureKeys = activeSpace.disabledFeatures;
 
-  const disabledFeatures = disabledFeatureKeys
-    .map((key) => features.find((feature) => feature.id === key))
-    .filter((feature) => typeof feature !== 'undefined') as Feature[];
+  const [enabledFeatures, disabledFeatures] = features.reduce(
+    (acc, feature) => {
+      if (disabledFeatureKeys.includes(feature.id)) {
+        return [acc[0], [...acc[1], feature]];
+      }
+      return [[...acc[0], feature], acc[1]];
+    },
+    [[], []] as [Feature[], Feature[]]
+  );
 
   const navLinks = capabilities.navLinks;
   const catalogueEntries = capabilities.catalogue;
   const managementItems = capabilities.management;
 
+  const enabledAppEntries = new Set(enabledFeatures.flatMap((ef) => ef.app ?? []));
+  const enabledCatalogueEntries = new Set(enabledFeatures.flatMap((ef) => ef.catalogue ?? []));
+  const enabledManagementEntries = enabledFeatures.reduce((acc, feature) => {
+    const sections = Object.entries(feature.management ?? {});
+    sections.forEach((section) => {
+      if (!acc.has(section[0])) {
+        acc.set(section[0], []);
+      }
+      acc.get(section[0])!.push(...section[1]);
```

Comment:
note: duplicate entries are technically possible here, but it seems tolerable given the lifetime and size of this array